File: ScaleformUI_Csharp/Menus/UIMenu/Items/UIMenuCheckboxItem.cs
using ScaleformUI.Elements;

namespace ScaleformUI.Menu
{
    public enum UIMenuCheckboxStyle
    {
        Cross,
        Tick
    }

    public class UIMenuCheckboxItem : UIMenuItem
    {
        private bool _checked;

        /// <summary>
        /// Triggered when the checkbox state is changed.
        /// </summary>
        public event ItemCheckboxEvent CheckboxEvent;

        public UIMenuCheckboxStyle Style { get; }

        /// <summary>
        /// Checkbox item with a toggleable checkbox.
        /// </summary>
        /// <param name="text">Item label.</param>
        /// <param name="check">Boolean value whether the checkbox is checked.</param>
        public UIMenuCheckboxItem(string text, bool check) : this(text, check, "")
        {
        }

        /// <summary>
        /// Checkbox item with a toggleable checkbox.
        /// </summary>
        /// <param name="text">Item label.</param>
        /// <param name="check">Boolean value whether the checkbox is checked.</param>
        /// <param name="description">Description for this item.</param>
        public UIMenuCheckboxItem(string text, bool check, string description) : this(text, UIMenuCheckboxStyle.Tick, check, description, SColor.HUD_Panel_light, SColor.HUD_White)
        {
        }

        /// <summary>
        /// Checkbox item with a toggleable checkbox.
        /// </summary>
        /// <param name="text">Item label.</param>
        /// <param name="style">CheckBox style (Tick or Cross).</param>
        /// <param name="check">Boolean value whether the checkbox is checked.</param>
        /// <param name="description">Description for this item.</param>
        public UIMenuCheckboxItem(string text, UIMenuCheckboxStyle style, bool check, string description) : this(text, style, check, description, SColor.HUD_Panel_light, SColor.HUD_White)
        {
        }

        /// <summary>
        /// Checkbox item with a toggleable checkbox.
        /// </summary>
        /// <param name="text">Item label.</param>
        /// <param name="style">CheckBox style (Tick or Cross).</param>
        /// <param name="check">Boolean value whether the checkbox is checked.</param>
        /// <param name="description">Description for this item.</param>
        /// <param name="mainColor">Main item color.</param>
        /// <param name="highlightColor">Highlight item color.</param>
        public UIMenuCheckboxItem(string text, UIMenuCheckboxStyle style, bool check, string description, SColor mainColor, SColor highlightColor) : base(text, description, mainColor, highlightColor)
        {
            Style = style;
            _checked = check;
            _itemId = 2;
        }


        /// <summary>
        /// Change or get whether the checkbox is checked.
        /// </summary>
        public bool Checked
        {
            get => _checked;
            set
            {
                _checked = value;
                if (Parent != null && Parent.Visible)
                    Parent.SendItemToScaleform(Parent.MenuItems.IndexOf(this), true);
                if (ParentColumn != null && ParentColumn.visible)
                    ParentColumn.SendItemToScaleform(ParentColumn.Items.IndexOf(this), true);
            }
        }

        public void CheckboxEventTrigger()
        {
            CheckboxEvent?.Invoke(this, Checked);
        }

        public override void SetRightBadge(BadgeIcon badge)
        {
            throw new Exception("UIMenuCheckboxItem cannot have a right badge.");
        }

        public override void SetRightLabel(string text)
        {
            throw new Exception("UIMenuCheckboxItem cannot have a right label.");
        }
    }
}

File: ScaleformUI_V8/src/menus/UIMenu/items/uimenucheckboxitem.ts
import { DelegateBuilder, IDelegate } from "helpers/eventemitter";
import { UIMenuItem } from "./uimenuitem";
import { SColor } from "elements/scolor";
import { BadgeStyle } from "elements/badge";
import { CheckboxItemChangeCallbackBuilder, UIMenuCheckboxItemChangeCallback } from "../emitters/emitters";

export enum UIMenuCheckboxStyle {
    Cross,
    Tick
}

export class UIMenuCheckboxItem extends UIMenuItem {
    private _checked: boolean = false;
    public Style: UIMenuCheckboxStyle = UIMenuCheckboxStyle.Cross;
    private _checkedEmitter = new CheckboxItemChangeCallbackBuilder();

    constructor(text:string, style:UIMenuCheckboxStyle, check:boolean, description:string, mainColor:SColor, highlightColor:SColor) {
        super(text, description, mainColor, highlightColor, SColor.HUD_White, SColor.HUD_Black)
        this.Style = style;
        this._checked = check;
        this._itemId = 2;
    }

    public onCheckboxEvent(delegate: UIMenuCheckboxItemChangeCallback) {
        this._checkedEmitter.add(delegate);
    }

    checkEmit() {
        this._checkedEmitter.toDelegate()(this, this._checked)
    }

    public set Checked(value:boolean){
        this._checked = value;
        if (this.Parent !== null && this.Parent.Visible && this.Parent.Pagination.IsItemVisible(this.Parent.Items.indexOf(this))) {
            ScaleformUI.Scaleforms._ui.callFunction("SET_INPUT_EVENT", 16, this.Parent.Pagination.GetScaleformIndex(this.Parent.Items.indexOf(this)), value);
        }
    }
    public get Checked():boolean{
        return this._checked;
    }

    public SetRightBadge(badge: BadgeStyle) {
        throw new Error("UIMenuCheckboxItem cannot have a right badge.");
    }
    
    public SetRightLabel(text: string) {
        throw new Error("UIMenuCheckboxItem cannot have a right label.");
    }
}